core/src/test/java/org/infinispan/stress/AsyncStoreStressTest.java

```diff
@@ -134,7 +147,7 @@ void stopMarshaller() throws InterruptedException {
    // ImmortalCacheEntry{key=key165168, value=ImmortalCacheValue {value=61456}}}
    // (Thread-194:) Expected state updated with key=key165168, value=61456
    // (Thread-200:) Expected state updated with key=key165168, value=60483
-   private LockContainer locks = new PerKeyLockContainer();
+   private LockContainer locks;
```

Comment:
I didn't write this test at all. I would rather just remove all the garbage tbh. But I wasn't trying to completely rewrite this test, just get it to work.

Reply:
So I change the name of `withStore` to `withKeyLock` but I am not sure how you want me to change the description for this method.

I don't quite understand your comment about that the lock is held while doing the store write. That is how ISPN works. If the write is enqueued then it will release before it is written to the store and if it is enqueued but was "full" then it has to wait until it can be written before releasing the lock.